### PR TITLE
Fix a bug in TimeSplitter::getWorkspaceIndexName().

### PR DESCRIPTION
**Pull request by bolotovskyr**
#### Description of work

Defect [EWM3075](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=3075)

This PR fixes a bug in `TimeSplitter::getWorkspaceIndexName()`. The function assumes (and `assert`s) that `numericalShift` is positive. However, it doesn't have to be. Therefore, we need to remove the assertion and make the function work with negative shifts as well.

#### To test:
Run this script:
```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

LoadEventNexus(Filename="/SNS/EQSANS/shared/sans-backend/data/ornl/sans/sns/eqsans/EQSANS_104088.nxs.h5", OutputWorkspace="load_tmp")

GenerateEventsFilter(
    InputWorkspace="load_tmp",
    OutputWorkspace="filter",
    InformationWorkspace="info",
    TimeInterval=None,
    LogName="SampleTemp",
    LogValueInterval=0.1,
)

FilterEvents(
    InputWorkspace="load_tmp",
    SplitterWorkspace="filter",
    OutputWorkspaceBaseName="EQSANS_104088",
    InformationWorkspace="info",
    FilterByPulseTime=True,
    GroupWorkspaces=True,
    OutputWorkspaceIndexedFrom1=True,
)
```
and check that the output `EQSANS_104088` workspaces have suffixes: `_1`,`_2`,`_3`.  Before the bugfix they were: `_199`,`_200`,`_201`.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: Framework/DataObjects/src/TimeSplitter.cpp [hunk 1]
```diff
@@ -206,8 +206,7 @@ std::string TimeSplitter::getWorkspaceIndexName(const int workspaceIndex, const
 
   // If numericalShift > 0, the caller will get back a shifted index.
```
Comment (jmborr) [inline Framework/DataObjects/src/TimeSplitter.cpp:207]: I don't think this comment makes sense, right? Can you adapt it?

**Review by jmborr — CHANGES_REQUESTED** [review]
would you do a system test using the events file and example that prompted all the discussion?

**Comment by bolotovskyr** [comment]
Could you suggest a name for the system test?  Should it be something `SANS...`?  Of course, here we would be testing specifically `FilterEvents` `OutputWorkspaceIndexedFrom1` property, i.e. whether the output workspaces are counted from 1. Interestingly, we don't have a unit test for that property either.

**Comment by jmborr** [comment]
> Could you suggest a name for the system test? Should it be something `SANS...`? Of course, here we would be testing specifically `FilterEvents` `OutputWorkspaceIndexedFrom1` property, i.e. whether the output workspaces are counted from 1. Interestingly, we don't have a unit test for that property either.

The `SplitterWorkspace` output by `GenerateEventsFilter` had its first index at 199, if I remember correctly. So something like `testLargeFirstIndex` plus a docstring providing a detailed explanation

File: Framework/DataObjects/src/TimeSplitter.cpp (hunk 1 above)
Comment (bolotovskyr) [reply Framework/DataObjects/src/TimeSplitter.cpp]: Done.

**Review by jmborr — APPROVED** [review]

**Pull request merged**